docs/deploying_module_template.md

```diff
@@ -0,0 +1,66 @@
+## Deploying `ModuleTemplate` with the Lifecycle Manager
```

Comment:
I'm wondering if we should have this in the README.md since it is kind of related to the development flows that are described there. At least we should link this document there. If you keep this file as it is, start with a H1 (#).